src/compiler/checker.ts

```diff
@@ -16085,6 +16095,18 @@ namespace ts {
                 }
             }
 
+            if (node.type) {

```

Comment:
You should be performing this check in *any* type literal, not just those in variable like declarations.

I think you should just move the check into `checkGrammarProperty`, and discriminate on whether you have a `PropertySignature` with an initializer.